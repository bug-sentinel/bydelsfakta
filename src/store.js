import Vue from 'vue';
import Vuex from 'vuex';
import router from './router';

import districts from './config/geoData/districts';
import bydeler from './config/allDistricts';

Vue.use(Vuex);

const state = {
  compareDistricts: false,
  districts: [],
  districtsGeo: districts,
};

const getters = {
  geoDistricts: state => {
    if (!state.compareDistricts && state.districts.length !== 0) {
      return { ...state.districtsGeo[`${bydeler.find(district => district.key === state.districts[0]).uri}`] };
    } else if (state.districts[0] === 'alle') {
      return { ...state.districtsGeo.oslo };
    }

    const features = state.districts.map(id =>
      state.districtsGeo.oslo.features.find(district => district.properties.id === parseInt(id))
    );

    return {
      ...state.districtsGeo.oslo,
      features,
    };
  },
};

const mutations = {
  ADD_DISTRICT(state, payload) {
    state.compareDistricts = true;
    state.districts = payload;
  },
  SELECT_DISTRICT(state, payload) {
    state.compareDistricts = false;
    state.districts = payload;
  },
  CLEAN_STATE(state) {
    state.compareDistricts = false;
    state.districts = [];
    state.districtsGeo = districts;
  },
};

const actions = {
  addDistrict({ commit }, payload) {
    const districts = payload.district.split('-');
    if (districts.length === 1) {
      if (districts[0] === 'alle') {
        commit('ADD_DISTRICT', districts);
      } else {
        const districtValue = bydeler.find(district => district.uri === districts[0]);
        const districtKey = bydeler.find(district => district.key === districts[0]);
        districtValue === undefined
          ? commit('ADD_DISTRICT', [districtKey.key])
          : commit('SELECT_DISTRICT', [districtValue.key]);
      }
    } else {
      commit('ADD_DISTRICT', districts);
    }

    if (payload.pushRoute) {
      router.currentRoute.params.topic === undefined
        ? router.push({ name: 'District', params: { district: districts.join('-') } })
        : router.push({
            name: 'Topic',
            params: { district: districts.join('-'), topic: router.currentRoute.params.topic },
          });
    }
  },
  cleanState({ commit }) {
    commit('CLEAN_STATE');
  },
};

export default new Vuex.Store({
  strict: true,
  state,
  getters,
  mutations,
  actions,
});
