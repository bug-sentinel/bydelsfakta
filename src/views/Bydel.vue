<template>
  <div class="main-container">
    <div class="main-container__cards">
      <card-image
        v-for="item in items"
        :key="item.value"
        :id="item.value"
        :kategori="item.options.kategori"
        :tema="item.options.tema"
        :bg-image="item.options.bgImage"
        :bg-color="item.options.bgColor"
        :txt-color="item.options.txtColor"
        :link="`${$route.path}/${item.value}`"
      />
    </div>
    <div class="main-container__map">
      <h3>Bydelskart</h3>
      <leaflet-map :district="geoDistricts" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import CardImage from '../components/CardImage.vue';
import LeafletMap from '../components/LeafletVue.vue';
import subpages from '../config/subpages';
import districts from '../config/geoData/districts';

export default {
  name: 'Bydel',
  components: {
    CardImage,
    LeafletMap,
  },

  props: {
    bydel: {
      type: String,
      required: true,
    },
  },

  computed: {
    ...mapGetters(['geoDistricts']),
  },

  data() {
    return {
      items: subpages,
      chosenDistrict: {},
      districts: districts,
    };
  },
};
</script>
