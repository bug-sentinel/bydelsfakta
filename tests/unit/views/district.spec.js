import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';

import Bydel from '../../../src/views/District.vue';
import setupI18n from '../../../src/i18n';
import store from '../../../src/store';

const i18n = setupI18n();

describe('Bydel', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    localVue.use(Vuex);
    const router = new VueRouter();
    wrapper = shallowMount(Bydel, {
      propsData: {
        district: 'bydel',
      },
      localVue,
      router,
      store,
      i18n,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders bydel-component and finds main-container class', () => {
    expect(wrapper.classes('main-container')).toBe(true);
  });

  test('renders component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
