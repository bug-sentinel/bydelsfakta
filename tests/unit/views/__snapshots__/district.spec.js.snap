// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bydel renders component correctly 1`] = `
<div
  class="main-container"
>
  <div
    class="main-container__cards"
  >
    <div
      class="topics-grid"
    >
      <v-category-stub
        bgcolor="rgb(255, 129, 116)"
        bgimage="/img/folkemengde.png"
        category="Befolkning"
        id="folkemengde"
        link="//folkemengde"
        topic="Folkemengde"
        txtcolor="rgb(245, 173, 165)"
      />
      <v-category-stub
        bgcolor="rgb(255, 129, 116)"
        bgimage="/img/alder.png"
        category="Befolkning"
        id="alder"
        link="//alder"
        topic="Alder"
        txtcolor="rgb(245, 173, 165)"
      />
      <v-category-stub
        bgcolor="rgb(255, 129, 116)"
        bgimage="/img/innvandring.png"
        category="Befolkning"
        id="innvandrerbefolkningen"
        link="//innvandrerbefolkningen"
        topic="Innvandrer&shy;befolkningen"
        txtcolor="rgb(245, 173, 165)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/husholdninger.png"
        category="Bomiljø"
        id="husholdninger"
        link="//husholdninger"
        topic="Husholdninger"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/trangboddhet.png"
        category="Bomiljø"
        id="trangboddhet"
        link="//trangboddhet"
        topic="Trangboddhet"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/eierform.png"
        category="Bomiljø"
        id="eierform"
        link="//eierform"
        topic="Eierform"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/kommunale-boliger.png"
        category="Bomiljø"
        id="kommunaleboliger"
        link="//kommunaleboliger"
        topic="Kommunale boliger"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/flytting.png"
        category="Bomiljø"
        id="flytting"
        link="//flytting"
        topic="Flytting"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/bygningstyper.png"
        category="Bomiljø"
        id="bygningstyper"
        link="//bygningstyper"
        topic="Bygningstyper"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(66, 248, 182)"
        bgimage="/img/boligpriser.png"
        category="Bomiljø"
        id="boligpriser"
        link="//boligpriser"
        topic="Boligpriser"
        txtcolor="rgb(199, 247, 201)"
      />
      <v-category-stub
        bgcolor="rgb(248, 198, 107)"
        bgimage="/img/levekaar.png"
        category="Levekår"
        id="levekaar"
        link="//levekaar"
        topic="Levekår"
        txtcolor="rgb(244, 206, 138)"
      />
    </div>
  </div>
   
  <div
    class="main-container__map"
  >
    <h3>
      Bydelskart
    </h3>
     
    <v-leaflet-stub
      district="[object Object]"
    />
  </div>
</div>
`;
